utxonursery.go

```diff
@@ -1002,16 +816,30 @@ func (u *utxoNursery) registerSweepConf(finalTx *wire.MsgTx,
 // to mark any mature channels as fully closed in channeldb.
 // NOTE(conner): this method MUST be called as a go routine.
 func (u *utxoNursery) waitForSweepConf(classHeight uint32,
-	kgtnOutputs []kidOutput, confChan *chainntnfs.ConfirmationEvent) {
+	output *kidOutput, resultChan chan sweep.Result) {
 
 	defer u.wg.Done()
 
 	select {
-	case _, ok := <-confChan.Confirmed:
+	case result, ok := <-resultChan:
 		if !ok {
-			utxnLog.Errorf("Notification chan closed, can't"+
-				" advance %v graduating outputs",
-				len(kgtnOutputs))
+			utxnLog.Errorf("Notification chan closed, can't" +
+				" advance graduating output")
+			return
+		}
+
+		// In case of a remote spend, still graduate the output. There
+		// is no way to sweep it anymore.
```

Comment:
No, it isn't reflected properly. But it isn't reported properly in the current master branch either. So for that matter, no regression.

The follow up PR of this one, #2000, is going to remove the nursery and its store output prefixes completely. The plan is to let contract resolvers report on their status and aggregate that into the report. There it's also easier to handle the remote spend case, because there is no persistent state involved.

I wouldn't want to extend nursery with new functionality now and remove it already in the next PR.